homeassistant/components/pcf8574/config_flow.py

```diff
@@ -76,16 +70,66 @@ async def async_step_user(
                 step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
             )
 
+        device_unique_id = (
+            f"{user_input[CONF_I2C_PORT_NUM]}_{user_input[CONF_I2C_ADDRESS]}"
+        )
+        await self.async_set_unique_id(device_unique_id)
+        self._abort_if_unique_id_configured()
+
         try:
+
             await validate_input(self.hass, user_input)
         except OSError:
             errors["base"] = "ioerror"
         except Exception:  # pylint: disable=broad-except
             _LOGGER.exception("Unexpected exception")
             errors["base"] = "unknown"
         else:
-            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
+            self.data = user_input
+            self.data[CONF_PINS] = []
+            return await self.async_step_pin()
+            # return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
 
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_pin(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle pin tep."""
+        errors: dict[str, Any] = {}
+        pin_num = len(self.data[CONF_PINS])
+        # first run len=0 -> pin_num = 0
+        # second run len=1 -> pin_num = 1
+        if user_input is None:
+            pin_name = f"pin_{pin_num}_name"
+            step_pin_data_schema = vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC
+                    ): cv.boolean,
+                    vol.Optional(CONF_PIN_NAME, default=pin_name): cv.string,
+                    vol.Optional(CONF_INPUT, default=False): cv.boolean,
+                    vol.Optional("add_another", default=False): cv.boolean,
```

Comment:
don't show that option if the user is configuring the last pin (you have a check in line 130)

Reply:
thanks. changed it to only show add_another on pin 0 to 6